citar-org.el

```diff
@@ -352,11 +352,23 @@ or citation-reference."
 
 (defun citar-org-list-keys ()
   "List citation keys in the org buffer."
-  (let ((org-tree (org-element-parse-buffer)))
-    (delete-dups
-     (org-element-map org-tree 'citation-reference
-       (lambda (r) (org-element-property :key r))
-       org-tree))))
+  (let* ((org-tree (org-element-parse-buffer))
```

Comment:
@yantar92 - could you be so kind as to tell me what I'm doing wrong here with `org-element-cache-map`?

I'm trying to get the list of all the `citation-reference` keys, just as in the then statement below it.

Reply:
`citation-reference` is not an element, it is an object.

Reply:
Got it; thanks!

Reply:
As I'm looking, is there any reason `org-cite-list-keys` shouldn't be available outside of export?

That's basically what we're recreating here.

Reply:
Full parsing is expensive.

Reply:
Basically, `org-cite-list-keys` is not going to be performant on large files. And it is not written to be working outside export scope.

Reply:
As I'm attempting to do here; is that a reasonable use of `org-element-cache-map`, that's likely to be performant?

Reply:
Oh wait, `citation` is also an object.

Reply:
`org-element-cache-map` does not currently support objects. Objects are not even cached. The best I can suggest for now is `re-search-forward` + `org-element-context`. 

Object caching and fast mapping over arbitrary buffer AST elements are in the works. They are not trivial though.